### PR TITLE
[2.6] Adding pandas serialization in standard XCom to the pass data guide and the custom XCom backend tutorial

### PR DESCRIPTION
**Pull request by TJaniF**
Almost missed this very neat PR. Standard XCom now serializes pd.DataFrame objects via pyarrow. I added that to the passing data guide, as well as made some changes to the wording in the `when to use xcom` section. 
In the custom XCom backend guide step 6 shows how to write a custom serialization method for pandas dataframes. I think eventually it might make sense to revisit that and change the step to serialize a different datatype, for the moment this is still probably going to be a common usecase and the method shown is via CSV, which people who want to be able to query their XCom stored in a blob storage will probably prefer to a pyarrow serialization. But let me know if you disagree and I should change the data type in step 6 :)

### CODE REVIEW

File: learn/custom-xcom-backends-tutorial.md [hunk 1]
```diff
@@ -694,7 +694,7 @@ To test your custom XCom backend you will run a simple DAG which pushes a random
 
 ## Step 6: Create a custom serialization method to handle Pandas dataframes
 
-A powerful feature of custom XCom backends is the possibility to create custom serialization and deserialization methods. This is particularly useful for handling objects that cannot be JSON-serialized. In this step, you will create a new custom XCom backend that can save the contents of a [Pandas](https://pandas.pydata.org/) dataframe as a CSV file.
+A powerful feature of custom XCom backends is the possibility to create custom serialization and deserialization methods. This is particularly useful for handling objects that cannot be JSON-serialized or if you want to serialize Pandas dataframes using a different method than `pyarrow`. In this step, you will create a new custom XCom backend that can save the contents of a [Pandas](https://pandas.pydata.org/) dataframe as a CSV file. Being serialized as CSV, the content of the XCom becomes accessible to CSV query tools like [csvq](https://mithrandie.github.io/csvq/).
```
Comment (kentdanas) [inline learn/custom-xcom-backends-tutorial.md:697]:
```suggestion
A powerful feature of custom XCom backends is the possibility to create custom serialization and deserialization methods. This is particularly useful for handling objects that cannot be JSON-serialized. In this step, you will create a new custom XCom backend that can save the contents of a [Pandas](https://pandas.pydata.org/) dataframe as a CSV file. Being serialized as CSV, the content of the XCom becomes accessible to CSV query tools like [csvq](https://mithrandie.github.io/csvq/).

Note that as of Airflow 2.6, Pandas dataframes can be serialized by XCom using `pyarrow`. This step is still useful if you are using an older version of Airflow, and to highlight the general process for creating a custom serialization method.
```
Maybe something like this? Trying to make this update a more natural integration, but let me know if you want to chat about how best to do that.

**Review by kentdanas — APPROVED** [review]
Thanks for catching this @TJaniF, this is a great update! A couple minor comments throughout on how we talk about this, but I'm open to opposing feedback and don't need to hold up the review process for any of it.

File: learn/custom-xcom-backends-tutorial.md [hunk 2]
```diff
@@ -19,8 +19,8 @@ Common reasons to use a custom XCom backend include:
...
-- Utilizing custom serialization and deserialization methods. By default, Airflow uses JSON serialization, which puts limits on the type of data that you can pass through XComs. Pickling is also available, but it has [known security implications](https://docs.python.org/3/library/pickle.html). A custom XCom backend allows you to implement your own serialization and deserialization methods.
-- Accessing XCom without accessing the metadata database.  
+- Utilizing custom serialization and deserialization methods. By default, Airflow uses JSON serialization and, as of Airflow 2.6, can serialize Pandas dataframes via [pyarrow](https://pypi.org/project/pyarrow/). To circumvent this limitation in local development there is the option to serialize data via [pickling](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#enable-xcom-pickling), which is often not suitable for production due to [known security implications](https://docs.python.org/3/library/pickle.html). A custom XCom backend allows you to implement your own serialization and deserialization methods.
```
[the remark this answers is not in the document]
Comment (TJaniF) [reply learn/custom-xcom-backends-tutorial.md]: I agree, it feels very wedged in 😅 . I think if someone comes to this tutorial directly and specifically wants a custom backend only to serialized pandas dfs of a size small enough that standard xcom would work... yes already unlikely and then they'd skip to step 6 and see the note. :)

**Review by TJaniF — COMMENTED** [review]
Thank you for the review @kentdanas ! I accepted all suggestions :)

@jwity and @lzdanski hopefully the last 2.6 PR, but it is a tiny one for a very useful feature 😅

File: learn/airflow-passing-data-between-tasks.md [hunk 3]
```diff
@@ -67,19 +67,21 @@ You can view your XComs in the Airflow UI by going to **Admin** > **XComs**. You
...
+When using the standard XCom backend, the size-limit for an XCom is determined by the metadata database you are using. Common sizes are: 
 
 - Postgres: 1 Gb
 - SQLite: 2 Gb
 - MySQL: 64 Kb
 
-You can see that these limits aren't very big. And even if you think your data might meet the maximum allowable limit, don't use XComs. Instead, use intermediary data storage, which is more appropriate for larger amounts of data.  
+You can see that these limits aren't very big. If you think your data passed via XCom might exceed the size of your metadata database, either use a custom XCom backend or [intermediary data storage](#intermediary-data-storage).
+
+The second limitation in using the standard XCom backend is that only certain types of data can be serialized. Airflow supports JSON serialization, as well as serialization of Pandas dataframes in version 2.6+, if you need to serialize other data types you can do so by using a [custom XCom backend](custom-xcom-backends-tutorial.md).
```
Comment (jwitz) [inline learn/airflow-passing-data-between-tasks.md:80]:
```suggestion
The second limitation in using the standard XCom backend is that only certain types of data can be serialized. Airflow supports JSON serialization, as well as Pandas dataframe serialization in version 2.6 and later. If you need to serialize other data types you can do so using a [custom XCom backend](custom-xcom-backends-tutorial.md).
```

File: learn/custom-xcom-backends-tutorial.md [hunk 4]
```diff
@@ -694,7 +694,9 @@ To test your custom XCom backend you will run a simple DAG which pushes a random
 
 ## Step 6: Create a custom serialization method to handle Pandas dataframes
 
-A powerful feature of custom XCom backends is the possibility to create custom serialization and deserialization methods. This is particularly useful for handling objects that cannot be JSON-serialized. In this step, you will create a new custom XCom backend that can save the contents of a [Pandas](https://pandas.pydata.org/) dataframe as a CSV file.
+A powerful feature of custom XCom backends is the possibility to create custom serialization and deserialization methods. This is particularly useful for handling objects that cannot be JSON-serialized. In this step, you will create a new custom XCom backend that can save the contents of a [Pandas](https://pandas.pydata.org/) dataframe as a CSV file. Being serialized as CSV, the content of the XCom becomes accessible to CSV query tools like [csvq](https://mithrandie.github.io/csvq/).
+
+Note that as of Airflow 2.6, Pandas dataframes can be serialized by the standard XCom backend using `pyarrow`. This step is still useful if you are using an older version of Airflow, and to highlight the general process for creating a custom serialization method.
```
Comment (jwitz) [inline learn/custom-xcom-backends-tutorial.md:699]:
```suggestion
Note that as of Airflow 2.6, Pandas dataframes can be serialized by the standard XCom backend using `pyarrow`. This step is still useful if you are using an older version of Airflow.
```

**Review by jwitz — APPROVED** [review]
LGTM! Just some minor wording suggestions

**Pull request merged**